validator_client/src/preparation_service.rs

```diff
@@ -163,6 +206,45 @@ impl<T: SlotClock + 'static, E: EthSpec> PreparationService<T, E> {
         Ok(())
     }
 
+    /// Starts the service which periodically sends connected beacon nodes validator registration information.
+    pub fn start_validator_registration_service(self, spec: &ChainSpec) -> Result<(), String> {
+        let log = self.context.log().clone();
+
+        info!(
+            log,
+            "Validator registration service started";
+        );
+
+        let spec = spec.clone();
+        let slot_duration = Duration::from_secs(spec.seconds_per_slot);
+
+        let executor = self.context.executor.clone();
+
+        let validator_registration_fut = async move {
+            loop {
+                if self.should_publish_at_current_slot(&spec) {
+                    //TODO(sean) we should probably make it so validator registrations happen well before the merge
```

Comment:
Since we should only publish unchanged signatures once per epoch, I think it's ok to remove this check. And I don't think people will use the builder flow until closer to the merge anyways. I think this is better than potentially flooding the relay an epoch before the merge